agent/engine/task_manager_unix_test.go

```diff
@@ -380,3 +388,67 @@ func TestEFSVolumeNextStateWithTransitionDependencies(t *testing.T) {
 		})
 	}
 }
+
+func TestCleanupExecEnabledTask(t *testing.T) {
+	cfg := getTestConfig()
+	ctrl := gomock.NewController(t)
+	mockTime := mock_ttime.NewMockTime(ctrl)
+	mockState := mock_dockerstate.NewMockTaskEngineState(ctrl)
+	mockClient := mock_dockerapi.NewMockDockerClient(ctrl)
+	mockImageManager := mock_engine.NewMockImageManager(ctrl)
+	defer ctrl.Finish()
+
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	taskEngine := &DockerTaskEngine{
+		ctx:          ctx,
+		cfg:          &cfg,
+		dataClient:   data.NewNoopClient(),
+		state:        mockState,
+		client:       mockClient,
+		imageManager: mockImageManager,
+	}
+	mTask := &managedTask{
+		ctx:                      ctx,
+		cancel:                   cancel,
+		Task:                     testdata.LoadTask("sleep5"),
+		_time:                    mockTime,
+		engine:                   taskEngine,
+		acsMessages:              make(chan acsTransition),
+		dockerMessages:           make(chan dockerContainerChange),
+		resourceStateChangeEvent: make(chan resourceStateChange),
+		cfg:                      taskEngine.cfg,
+	}
+	mTask.Task.ExecCommandAgentEnabled = true
+	mTask.SetKnownStatus(apitaskstatus.TaskStopped)
+	mTask.SetSentStatus(apitaskstatus.TaskStopped)
+	container := mTask.Containers[0]
+	dockerContainer := &apicontainer.DockerContainer{
+		DockerName: "dockerContainer",
+	}
+	tID, _ := mTask.Task.GetID()
+	removeAll = func(path string) error {
```

Comment:
I worry that this function is invoked in a goroutine. If this is the case, this might or might not be executed before the test ends. 
Can we change a boolean variable here and do the assertion at the end of the test instead?

Reply:
this is not invoked in a goroutine. This just replaces os.RemoveAll to a custom function for this unit test